script/pathfinder.lua

```diff
@@ -0,0 +1,293 @@
+local cust_lib = require("script/custom_lib")
+local collision_mask_util_extended = require("script/collision-mask-util-control")
+local debug_lib = require("script/debug_lib")
+
+-------------------------------------------------------------------------------
+--  Init
+-------------------------------------------------------------------------------
+
+local clean_linear_path_enabled = true
+local clean_path_steps_enabled = false
+local clean_path_steps_distance = 5
+local non_colliding_position_accuracy = 0.5
+
+local pathfinder = {}
```

Comment:
This sounds like it decentralizes the annotations file and makes it redundant?

Reply:
its not supposed to be centralized in one file for the most part. annotations should be in the file where the things get defined.

Only the annotations for the global variables should be centralized somewhere since there is no other way to do them at the moment.